### PR TITLE
Add notebooks to testing

### LINKED ISSUE #20: Notebooks in pytest
Jupyter notebooks are the main use case for our birds. We should experiment with adding the notebooks to the `pytest` suite.

### PR DESCRIPTION
**Pull request by nikola-rados**
# Changes
- Updates and moves original demo notebook
- Updates `Makefile` commands
- Mounts storage in `docker-compose` files

Resolves #20.

# Notes
Originally, we wanted the notebooks to be included in the `pytest` suite. However, a recent update to the `Makefile` introduced some commands that would run the notebooks as a suite for you using `nbval`. All we had to do was hook in our notebook and we were good to go. 

One slight complication was the password for the `connection_string` could not be directly used in the notebook. To get around this, the `Makefile` will create a `credentials.json` file which will store the secret. There is a dummy value setup by this process and the user must manually add the correct value. This has to be done before the notebooks will work correctly. The file is in the `.gitignore` so there should be no worry about exposing the passphrase publicly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@ venv/
 *.sqlite
 .ipynb_checkpoints/
 .Rhistory
+credentials.json
```
Comment (sum1lim) [inline .gitignore:8]: Good catch! It's nice that you thought of this.

**Review by cairosanders — APPROVED** [review]
```
docs/source/notebooks/example.ipynb::Cell 0
  /tmp/sandpiper-venv/lib/python3.8/site-packages/jupyter_client/manager.py:66: DeprecationWarning: KernelManager._kernel_spec_manager_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
    def _kernel_spec_manager_changed(self):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```
I just got this warning but otherwise all the tests passed!

File: Makefile [hunk 2]
```diff
@@ -132,7 +138,7 @@ notebook-sanitizer:
 .PHONY: test-notebooks
 test-notebooks: notebook-sanitizer
 	@echo "Running notebook-based tests"
-	@bash -c "env WPS_URL=$(WPS_URL) pytest --nbval --verbose $(CURDIR)/docs/source/notebooks/ --sanitize-with $(CURDIR)/docs/source/output-sanitize.cfg --ignore $(CURDIR)/docs/source/notebooks/.ipynb_checkpoints"
+	@bash -c "source $(VENV)/bin/activate && env WPS_URL=$(WPS_URL) pytest --nbval --verbose $(CURDIR)/docs/source/notebooks/ --sanitize-with $(CURDIR)/docs/source/output-sanitize.cfg --ignore $(CURDIR)/docs/source/notebooks/.ipynb_checkpoints"
```
Comment (sum1lim) [inline Makefile:141]: I don't see the commands are being executed when running `make` command. Also, I don't understand what the commands do. Could you elaborate?

File: Makefile [hunk 3]
```diff
@@ -74,6 +76,10 @@ status: venv
 	@echo "Show status ..."
 	@-bash -c "${VENV}/bin/$(APP_NAME) status"
 
+.PHONY: credentials
+credentials:
+	test -f credentials.json || echo "{\n\t\"connection_string\": \"not_set\"\n}" > credentials.json
```
Comment (sum1lim) [inline Makefile:81]: Passing all the tests. :+1:

File: docker-compose.yml [hunk 4]
```diff
@@ -5,7 +5,8 @@ services:
     image: pcic/sandpiper
     ports:
       - "8101:5003"
-
+    volumes:
+      - /storage/data/climate/downscale/BCCAQ2/:/storage/data/climate/downscale/BCCAQ2/
```
Comment (sum1lim) [inline docker-compose.yml:9]: I wonder what this line is for. Would be appreciated if you could teach me.

**Review by sum1lim — APPROVED** [review]
LGTM! Demo and `make` command are working great.

File: Makefile (hunk 2 above)
Comment (nikola-rados) [reply Makefile:141]: The command is mostly setup. Setting up the environment and sanitizing the notebook before running `pytest` with `nbval` to run the notebook.

File: docker-compose.yml (hunk 4 above)
Comment (nikola-rados) [reply docker-compose.yml:9]: I discovered that our deployed instance of `sandpiper` was unable to correctly resolve the rules as it did not have access to the data. This just supplies the data as a volume that can be targeted using the same directory structure the database anticipates.

**Pull request merged**